plugins/debug.py

```diff
@@ -0,0 +1,323 @@
+        return
+
+    def dump_dnf_config_info(self, fobj):
+        var = self.base.conf.substitutions
+        plugins = ",".join([p.name for p in self.base.plugins.plugins])
+        fobj.write("%%%%DNF INFO\n")
+        fobj.write("  arch: %s\n" % var['arch'])
+        fobj.write("  basearch: %s\n" % var['basearch'])
+        fobj.write("  releasever: %s\n" % var['releasever'])
+        fobj.write("  dnf ver: %s\n" % dnf.const.VERSION)
+        fobj.write("  enabled plugins: %s\n" % plugins)
+        fobj.write("  global excludes: %s\n" % ",".join(self.base.conf.exclude))
+        return
+
+    def dump_rpm_problems(self, fobj):
+        fobj.write("%%%%RPMDB PROBLEMS\n")
+        (missing, conflicts) = rpm_problems(self.base)
+        fobj.writelines(["Package %s requires %s\n" % (unicode(pkg), unicode(req))
+                         for (req, pkg) in missing])
+        fobj.writelines(["Package %s conflicts with %s\n" % (unicode(pkg),
+                                                             unicode(conf))
+                         for (conf, pkg) in conflicts])
+
+
+    def dump_packages(self, fobj, load_repos):
+        self.base.fill_sack(load_system_repo=True,
+                            load_available_repos=load_repos)
+        q = self.base.sack.query()
+        # packages from rpmdb
+        fobj.write("%%%%RPMDB\n")
+        for p in sorted(q.installed()):
+            fobj.write('  %s\n' % pkgspec(p))
+
+        if not load_repos:
+            return
+
+        fobj.write("%%%%REPOS\n")
+        available = q.available()
+        for repo in sorted(self.base.repos.iter_enabled(), key=lambda x: x.id):
+            try:
+                url = None
+                if repo.metalink is not None:
+                    url = repo.metalink
+                elif repo.mirrorlist is not None:
+                    url = repo.mirrorlist
+                elif len(repo.baseurl) > 0:
+                    url = repo.baseurl[0]
+                fobj.write('%%%s - %s\n' % (repo.id, url))
+                fobj.write('  excludes: %s\n' % ','.join(repo.exclude))
+                for po in sorted(available.filter(reponame=repo.id)):
+                    fobj.write('  %s\n' % pkgspec(po))
+
+            except dnf.exceptions.Error as e:
+                fobj.write("Error accessing repo %s: %s\n" % (repo, str(e)))
+                continue
+        return
+
+    def dump_rpmdb_versions(self, fobj):
+        fobj.write("%%%%RPMDB VERSIONS\n")
+        version = self.base.sack.rpmdb_version(self.base.yumdb)
+        fobj.write('  all: %s\n' % version)
+        return
+
+
+class DebugRestoreCommand(dnf.cli.Command):
+
+    aliases = ['debug-restore']
+    summary = _('restore packages recorded in debug-dump file')
+    usage = '[%s] [%s]' % (_('OPTIONS'), _('KEYWORDS'))
+
+    def run(self, args):
+        """Execute the command action here."""
+
+        parser = dnfpluginscore.ArgumentParser(self.aliases[0])
```

Comment:
please make static function for parsing args as in other plugin: https://github.com/rpm-software-management/dnf-plugins-extras/blob/master/plugins/repomanage.py#L54